docs/src/main/paradox/s3.md

```diff
@@ -100,19 +100,72 @@ Java
 
 ### Java examples with custom headers
 
-#### With `ServerSideEncryption`
-
-```java
-final S3Client client = new S3Client(settings, system(), materializer);
-final CompletionStage<MultipartUploadResult> resultCompletionStage = client.multipartCopy(bucket(), bucketKey(), 
-                            targetBucket(), targetBucketKey(), S3Headers.empty(), ServerSideEncryption.AES256$.MODULE$);
-```
-
-### With `CannedAcl`
-
 ```java
-final CompletionStage<MultipartUploadResult> resultCompletionStage = client.multipartCopy(bucket(), bucketKey(), 
-                            targetBucket(), targetBucketKey(), S3Headers.apply(CannedAcl.Private$.MODULE$), null);
+import java.util.concurrent.CompletionStage;
+import akka.actor.ActorSystem;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.alpakka.s3.MemoryBufferType;
+import akka.stream.alpakka.s3.S3Settings;
+import akka.stream.alpakka.s3.acl.CannedAcl;
+import akka.stream.alpakka.s3.impl.ListBucketVersion2;
+import akka.stream.alpakka.s3.impl.S3Headers;
+import akka.stream.alpakka.s3.impl.ServerSideEncryption;
+import akka.stream.alpakka.s3.javadsl.MultipartUploadResult;
+import akka.stream.alpakka.s3.javadsl.S3Client;
+import akka.testkit.javadsl.TestKit;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.AwsRegionProvider;
+
+public class Test extends TestKit {
```

Comment:
Please put the code for the documentation in the tests and import the snippets via Parardox' @@snip support.
You may use the same snippet marker at several places in one file and Paradox will put it together in the order they appear, that way you can pull in correct Java imports from the source.